api/models/user.js

```diff
@@ -1,47 +1,85 @@
-            type: 'string', required: false, columnName: 'fullname', columnType: 'varchar(45)'
-        },
-        bio:{
-            type: 'string', required: false, columnName: 'bio', columnType: 'varchar(200)'
-        },
-        location:{
-            type: 'string', required: false, columnName: 'location', columnType: 'varchar(20)'
-            
-        },
-        isvisible:{
-            type: 'boolean', columnName: 'isvisible'
-        },
-        languages:{
-            type:'json', columnName: 'languages', defaultsTo: null, columnType: 'JSON'
-        },
-        email: {
-            type: 'string', columnName: 'email', unique: true, required: true, columnType: 'varchar(45)'
-        },
-        password: {
-            type: 'string', columnName: 'password', required: true, columnType: 'varchar(20)'
-        },
-        workinfo: {
-            collection: 'workinfo',
-            via: 'user'
+  attributes: {
+    id: {
+      type: 'number',
+      columnName: 'iduser',
+      columnType: 'int',
+      required: false,
+      autoIncrement: true,
+      unique: true,
+    },
+    username: {
+      type: 'string',
+      required: false,
+      columnName: 'username',
+      columnType: 'varchar(20)',
+    },
+    fullname: {
+      type: 'string',
+      required: false,
+      columnName: 'fullname',
+      columnType: 'varchar(45)',
+    },
+    bio: {
+      type: 'string',
+      required: false,
+      columnName: 'bio',
+      columnType: 'varchar(200)',
+    },
+    location: {
+      type: 'string',
+      required: false,
+      columnName: 'location',
+      columnType: 'varchar(20)',
+    },
+    isvisible: {
+      type: 'boolean',
+      columnName: 'isvisible',
+    },
+    languages: {
+      type: 'json',
+      columnName: 'languages',
+      defaultsTo: null,
+      columnType: 'JSON',
+    },
+    email: {
+      type: 'string',
+      columnName: 'email',
+      unique: true,
+      required: true,
+      columnType: 'varchar(45)',
+      isEmail:true
+    },
+    password: {
+      type: 'string',
+      columnName: 'password',
+      required: true,
+      columnType: 'varchar(200)',
+      encrypt: true,
+    },
+    workinfo: {
+      collection: 'workinfo',
+      via: 'user',
+    },
+    taskscreated: {
+      collection: 'tasks',
+      via: 'createdby',
+    },
+    workingon: {
+      collection: 'tasks',
+      via: 'takenby',
+    },
+  },
 
-        }, taskscreated: {
-            collection: 'tasks',
-            via: 'createdby'
-        },
-        workingon: {
-            collection: 'tasks',
-            via: 'takenby'
-        }
+  customToJSON: function () {
+    // Return a shallow copy of this record with the password and ssn removed.
+    return _.omit(this, ['password']);
+  },
 
-    }
-}
\ No newline at end of file
+  findByCredentials: async (email, password) => {
+    const user = await User.findOne({ email }).decrypt();
+    if (!user) throw new Error('Unable to connect');
```

Comment:
a throw will expect to be used by a try catch when this function is invoked. Instead of this make a dictionary that contains response names given by the specific case your handling. This will also make it harder for people to use this method in the future since they will have to read the definition of your method.